k8s/production/custom/runner-node-certificate-controller/cron-jobs.yaml

```diff
@@ -0,0 +1,201 @@
+# CronJob to remove certificate-pending taints from ready runner nodes
+# Fixes CSR timing issues that cause GitLab CI job failures
+---
+apiVersion: batch/v1
+kind: CronJob
+metadata:
+  name: runner-node-certificate-controller
+  namespace: custom
+  labels:
+    app: runner-node-certificate-controller
+    component: certificate-controller
+    purpose: csr-timing-fix
+spec:
+  schedule: "*/1 * * * *"  # Run every minute
+  concurrencyPolicy: Forbid  # Don't run concurrent jobs
+  successfulJobsHistoryLimit: 3
+  failedJobsHistoryLimit: 3
+  jobTemplate:
+    spec:
+      backoffLimit: 2
+      template:
+        metadata:
+          labels:
+            app: runner-node-certificate-controller-job
+        spec:
+          serviceAccountName: runner-node-certificate-controller
+          restartPolicy: OnFailure
+          nodeSelector:
+            spack.io/node-pool: base
+          containers:
+          - name: runner-node-certificate-controller
+            image: bitnami/kubectl:latest
+            command:
+            - /bin/bash
+            - -c
+            - |
+              set -e
```

Comment:
```suggestion
              set -eo pipefail
```

or better yet
```suggestion
              set -euo pipefail
```

Reply:
Thanks. I haven't run it through shellcheck/closely audited it beyond just making sure it does what I want it to. (I'm sure the overuse of emojis hints at where most of it came from :smiley:)